README.md

```diff
@@ -1,16 +1,201 @@
+typically have a `lib` folder and a `spec` folder and some files like
+`README.md`:
 
-## SWBAT 2
+```text
+├── lib
+│   └── ruby_file.rb
+└── spec
+    ├── ruby_file_spec.rb
+    └── spec_helper.rb
+├── Gemfile
+├── README.md
+```
+
+As we expand our understanding of programming in Ruby and start to work with
+larger, more complex applications, we need to add a few more pieces to this. In
+an upcoming lesson, we'll take a closer look at the `Gemfile` and what that
+does. For this lesson, though, we're going to take a look at some other folders
+and file structures you'll see in Ruby labs going forward. First, we're going to
+look at requiring files.
+
+If we define two classes in the same file, we can interact with both classes
+immediately after they are defined. Imagine if the following 'belongs-to' /
+'has-many' example between `Garden` and `Plant` classes:
+
+```ruby
+class Garden
+  attr_accessor :name
+
+  def initialize(name:)
+    @name = name
+  end
+
+  def plants
+    Plant.all.select do |plant|
+      plant.garden == self
+    end
+  end
+end
+
+class Plant
+  attr_accessor :garden, :name
+
+  @@all = []
+
+  def initialize(name:)
+    @name = name
+    @@all << self
+  end
+
+  def self.all
+    @@all
+  end
+end
+
+lawn = Garden.new(name: 'Front Lawn')
+
+basil = Plant.new(name: 'Basil')
+basil.garden = lawn
+
+cucumber = Plant.new(name: 'Cucumber')
+cucumber.garden = lawn
+
+p lawn.plants
+# => [#<Plant:0x00007fa4440ab0c8 @name="Basil", @garden=#<Garden:0x00007fa4440997b0 @name="Front Lawn">>, #<Plant:0x00007fa4440b8bd8 @name="Cucumber", @garden=#<Garden:0x00007fa4440997b0 @name="Front Lawn">>]
+```
+
+> **Note**: This code is available in `lib/example.rb` and can be tested by
+running `ruby lib/example.rb`.
+
+In the above code, we can call both `Garden` and `Plant` at the end because both
+classes have been defined in the file. Notice, too, that `Garden` includes a
+method, `plants`, that calls `Plant.all`. `Garden` _needs_ to know about the
+`Plant` class, and with the setup above, it does.
+
+If, however, we define these classes in separate files, they won't automatically
+know about each other or have access to one another. In the `lib` folder,
+`Plant` and `Garden` are separated into their own files, `plant.rb` and
+`garden.rb`. Below the `Garden` class, the same code from the previous example
+is written:
+
+```ruby
+# lib/garden.rb
+
+# ...
+
+lawn = Garden.new(name: 'Front Lawn')
+
+basil = Plant.new(name: 'Basil')
+basil.garden = lawn
+
+cucumber = Plant.new(name: 'Cucumber')
+cucumber.garden = lawn
+
+p lawn.plants
+```
+
+At the moment, if we try to run the file (`ruby lib/garden.rb`), we get an error
+regarding an **uninitialized constant Plant (NameError)**. For one file to have
+access to code written in another file, we need to tell Ruby to _require_ that
+other file.
+
+## Define `require_relative` and `require`
+
+By requiring a file, we are telling Ruby, 'go load the code exists in this file.'
```

Comment:
'go load the code exists in this file.' Is this supposed to be 'go load the code that exists in this file.'?